app/assets/src/components/views/phylo_tree/PhyloTreeDownloadButton.jsx

```diff
@@ -1,28 +1,55 @@
 import React from "react";
 import PropTypes from "prop-types";
+import SvgSaver from "svgsaver";
+
 import DownloadButtonDropdown from "../../ui/controls/dropdowns/DownloadButtonDropdown";
 
 class PhyloTreeDownloadButton extends React.Component {
   constructor(props) {
     super(props);
 
-    this.allOptions = [
+    this.dataOptions = [
       { text: "VCF", value: "vcf" },
       { text: "SNP annotations", value: "snp_annotations" }
     ];
+    this.imageOptions = [
+      { text: "SVG", value: "svg" },
+      { text: "PNG", value: "png" }
+    ];
     this.download = this.download.bind(this);
+    this.svgSaver = new SvgSaver();
   }
 
   download(option) {
-    location.href = `/phylo_trees/${
-      this.props.tree.id
-    }/download?output=${option}`;
+    if (this.dataOptions.map(o => o.value).includes(option)) {
+      location.href = `/phylo_trees/${
+        this.props.tree.id
+      }/download?output=${option}`;
+      return;
+    } else if (option == "svg") {
+      // TODO (gdingle): filename per tree?
+      this.svgSaver.asSvg(this.getNode(), "phylo_tree.svg");
+      return;
+    } else if (option == "png") {
+      // TODO (gdingle): filename per tree?
+      this.svgSaver.asPng(this.getNode(), "phylo_tree.png");
+      return;
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(`Bad download option: ${option}`);
+    }
+  }
+
+  // TODO (gdingle): should we pass in a reference with React somehow?
```

Comment:
I would add a callback and have the viz take care of saving it.
In general, if you need properties/objects/references that do not make much sense in a particular component, callbacks are normally the answer.